tests/foreman/cli/test_host.py

```diff
@@ -1852,163 +1848,34 @@ def test_positive_install_package_via_rex(
-        :param auto_attach: boolean to indicate whether to register with
-            auto-attach option, in case of registration with activation key a
-            command is launched
-        :return: the registration result
-        """
-        activation_key = activation_key or self.ak
-
-        if lce:
-            result = self.client.register_contenthost(
-                self.org.name,
-                lce=f'{self.host_lce.name}/{self.content_view.name}',
-                auto_attach=auto_attach,
-            )
-        else:
-            result = self.client.register_contenthost(
-                self.org.name, activation_key=activation_key.name
-            )
-            if auto_attach and self.client.subscribed:
-                result = self.client.run('subscription-manager attach --auto')
-
-        if self.client.subscribed and enable_repo:
-            self.client.enable_repo(self.repository_id)
-
-        return result
-
-    def _make_activation_key(self, add_subscription=False):
-        """Create a new activation key
-
-        :param add_subscription: boolean to indicate whether to add the default
-            subscription to the created activation key
-        :return: the created activation key
-        """
-        activation_key = entities.ActivationKey(
-            content_view=self.content_view,
-            organization=self.org,
-            environment=self.host_lce,
-            auto_attach=False,
-        ).create()
-        if add_subscription:
-            activation_key.add_subscriptions(data={'subscription_id': self.default_subscription_id})
-        return activation_key
-
-    def _host_subscription_register(self, request):
-        """Register the subscription of client as a content host consumer"""
-
-        @request.addfinalizer
-        def _cleanup():
-            # check whether the client was not unregistered in _register_client method
-            if (
-                datetime.utcnow().strftime("%Y-%m-%d")
-                in Host.info({'name': self.client.hostname})['subscription-information'][
-                    'registered-at'
-                ]
-                or (datetime.utcnow() - timedelta(days=1)).strftime("%Y-%m-%d")
-                in Host.info({'name': self.client.hostname})['subscription-information'][
-                    'registered-at'
-                ]
-            ):
-                Host.subscription_unregister({'host': self.client.hostname})
-
-        Host.subscription_register(
-            {
-                'organization-id': self.org.id,
-                'content-view-id': self.content_view.id,
-                'lifecycle-environment-id': self.host_lce.id,
-                'name': self.client.hostname,
-            }
-        )
+    ).create()
+    activation_key.add_subscriptions(data={'subscription_id': default_subscription.id})
+    return activation_key
 
 
 # -------------------------- HOST SUBSCRIPTION SUBCOMMAND SCENARIOS -------------------------
```

Comment:
Feel free to group the tests into classes if you would like to better organize them within the module - or split the module up! Your call - and not in this PR.